plugins/network-elements/nuage-vsp/src/com/cloud/network/guru/NuageVspGuestNetworkGuru.java

```diff
@@ -470,14 +468,14 @@ public void reserve(NicProfile nic, Network network, VirtualMachineProfile vm, D
             HostVO nuageVspHost = _nuageVspManager.getNuageVspHost(network.getPhysicalNetworkId());
             VspNetwork vspNetwork = _nuageVspEntityBuilder.buildVspNetwork(vm.getVirtualMachine().getDomainId(), network);
 
-            if (vm.getType() == VirtualMachine.Type.DomainRouter && vspNetwork.getVirtualRouterIp().equals("null")) {
+            boolean isUpgradeNetworkOffering = vm.getType() == VirtualMachine.Type.DomainRouter && vspNetwork.getVirtualRouterIp()
```

Comment:
please explain; if the vm is a router and the network has no ip, then we are upgrading a network offering? I am missing the logic.

Reply:
@DaanHoogland if a VR is spinned in a network that currently has no virtual router ip (vspNetwork.getVRip==null), it means that there was no virtual router before and that the network offering is changed.

Reply:
isn't it possible that the old offering had a virtual router as well? it seems to me that only oldOffering.equals(newOffering) is a proper test here.

Reply:
@sgoeminn how about this question?

Reply:
@DaanHoogland if the old offering had a VR and the new has, in that case the broadcast uri does not need to be updated because the VR ip does not change.

Reply:
How about when more then one provider type is used in an install? Can we guarantee this call is only called when? either the name isUpgradeNetworkOffering is deceiving or the test is incomplete. Is this meant to be a 'only if migrating from VR to nuage implementation' check?

Reply:
You are right, the name of the boolean is a bit misleading. It is in fact a VRAddedToNuage check (for example when there was no user data in the network and a user wants to add the VR as user data provider). I will change it tomorrow.

Reply:
@DaanHoogland I changed it!